### PR TITLE
Helm: Add alibabacloud to isUsingObjectStorage.

### LINKED ISSUE #11945: [Helm] alibabacloud storage type is not supported
**Describe the bug**
The Loki Helm chart does not allow using `alibabacloud` in `.Values.loki.storage.type`.

**To Reproduce**
1. Configure Helm value `.Values.loki.storage.type` to be `alibabacloud`
2. Run `helm template` or `helm install`

**Expected behavior**
The Helm chart allows using the value and injects it in Helm templates where referenced.

**Environment:**
 - Infrastructure: Kubernetes
 - Deployment tool: helm

**Screenshots, Promtail config, or terminal output**
Output from `helm template`:
```txt
Error: execution error at (context/charts/loki/templates/validate.yaml:26:4): Cannot run Scalable targets (backend, read, write) without an object storage backend.
```

### PR DESCRIPTION
**Pull request by lindeskar**
**What this PR does / why we need it**:
Allow setting `alibabacloud` in `.Values.loki.storage.type`.

**Which issue(s) this PR fixes**:
Fixes #11945

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by MichelHollands** [comment]
@lindeskar Thanks for your contribution, Can you update the version in all the places mentioned in the Helm chart checklist item? Once that is done I'll merge it (if there are no other issues).

**Comment by lindeskar** [comment]
I thought I had. 😄 Fixed now!
Let me know if something else is missing.

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**
